allocator.go

```diff
@@ -120,16 +194,10 @@ func New(ranges []Range, nodes []Node, opts ...Option) *Allocator {
 	}
 
 	// build a convenience id-to-struct mapping for ranges.
-	idToRangeMap := make(map[RangeID]Range)
-	for _, r := range ranges {
-		idToRangeMap[r.id] = r
-	}
+	idToRangeMap := createRangeMap(ranges)
 
 	// build a convenience id-to-struct mapping for nodes.
-	idToNodeMap := make(map[NodeID]Node)
-	for _, n := range nodes {
-		idToNodeMap[n.id] = n
-	}
+	idToNodeMap := createNodeMap(nodes)
```

Comment:
We would also need to remove the `nodes/ranges` being ingested here and instead coming into play inside the `Allocate` method.